runtime/reference/security_model.md

```diff
@@ -0,0 +1,184 @@
+  analyzed require runtime read permissions.
+- Files inside of a `node_modules/` directory are allowed to be read by default.
+
+When fetching modules from the network, or when transpiling code from TypeScript
+to JavaScript, Deno uses the file system as a cache. This means that file system
+resources like storage space can be consumed by Deno even if the user has not
+explicitly granted read/write permissions.
+
+### Network access
+
+By default, executing code can not make network requests, open network listeners
+or perform DNS resolution. This includes making HTTP requests, opening TCP/UDP
+sockets, and listening for incoming connections on TCP or UDP.
+
+During module loading, Deno can load modules from the network. By default Deno
+allows loading modules from the following locations using both static and
+dynamic imports:
+
+- `https://deno.land/`
+- `https://jsr.io/`
+- `https://esm.sh/`
+- `https://raw.githubusercontent.com`
+- `https://gist.githubusercontent.com`
+
+These locations are trusted "public good" registries that are not expected to
+enable data exfiltration through URL paths.
+
+In addition Deno allows importing any NPM package through `npm:` specifiers.
+
+Deno also sends requests to `https://dl.deno.land/` at most once a day to check
+for updates to the Deno CLI. This can be disabled using
+`DENO_NO_UPDATE_CHECK=1`.
+
+### Environment variables
+
+By default, executing code can not read or write environment variables. This
+includes reading environment variables, and setting new values.
+
+Deno reads certain environment variables on startup, such as `DENO_DIR` and
+`NO_COLOR` (see `deno help` for the full list).
+
+The value of the `NO_COLOR` environment variable is visible to all code running
+in the Deno runtime, regardless of whether the code has been granted permission
+to read environment variables.
+
+### Subprocesses
+
+Code executing inside of a Deno runtime can not spawn subprocesses by default,
+as this would constitute a violation of the principle that code can not escalate
+it's privileges without user consent.
+
+Deno provides a mechanism for executing subprocesses, but this requires explicit
+permission from the user. This is done by passing the `--allow-run` flag to the
+`deno` command.
+
+When running `deno install` with the `--allow-scripts` flag, post-install
+scripts for npm packages will be executed as a subprocess.
+
+### FFI
+
+Deno provides a mechanism for executing code written in other languages, such as
+Rust, C, or C++, from within a Deno runtime. This is done using the
+`Deno.dlopen` API, which can load shared libraries and call functions from them.
+
+By default, executing code can not use the `Deno.dlopen` API, as this would
+constitute a violation of the principle that code can not escalate it's
+privileges without user consent.
+
+In addition to `Deno.dlopen`, FFI can also be used via Node-API (NAPI) native
+addons. These are also not allowed by default.
```

Comment:
I think this should include "how to allow them" via the flag. Slightly worried that search engine summarization will assume that it's just not allowed